core/src/main/scala/kafka/utils/KafkaScheduler.scala

```diff
@@ -88,9 +89,14 @@ class KafkaScheduler(val threads: Int,
   
   override def shutdown() {
     debug("Shutting down task scheduler.")
-    this synchronized {
-      if(isStarted) {
-        executor.shutdown()
+    // Synchronize on the shutdownLock to avoid NullPointerException when multiple threads are shutting down scheduler
+    // at same time.
+    shutdownLock synchronized {
+      if (isStarted) {
+        this synchronized {
+          this.executor.shutdown()
```

Comment:
Actually you probably do need to (but unnecessary if we are handling rejected executions in schedule - which we are not)